### ISSUE TITLE
Questions about the Start Menu, and some issues

### ISSUE THREAD

**Issue by someguy16** [rozniak/xfce-winxp-tc#405]
![startmenu](https://github.com/user-attachments/assets/db00457d-ffbb-4f1c-9f06-a4dd33b0f7a6)

1. Abstractile, Binary Horizon, Binary Ring? Are those screensavers? I ran them, I didn't see anything but I saw their processes in task manager. When I went to log out, instead of seeing the split second black screen I saw a bunch of colors that looked like graphical corruption (makes sense if I was running a dozen screensavers at once). I didn't have these in the start menu when I installed this in a VM once before, but another issue had a screenshot with similar entries in the start menu.

2. Any way I can change what applications are in the Start Menu?

Some issues:

The start menu can't seem to find the icon for Seamonkey, a browser I am using for my default email client (though the browser is based on Firefox ~56 the last update it received was on September of this year, so I don't think it's ancient and what's causing the missing icon).

The taskbar doesn't adjust when changing the resolution, it remains where it was before.

Changing between themes can cause the All Program menu(s) to keep jittering up and down (and/or add arrows for scrolling despite the menus having enough space). It usually takes just opening and reopening the menu to fix it, but it can be more stubborn depending on what theme was changed and what theme was applied (it doesn't fix in one go when changing from Windows XP Blue to Professional).

The start menu's usually offset when opening it for the first time after changing the theme (and it fixes itself when you close it) but that's _really_ minor and you probably knew that already.

Sorry if you knew some of these things, I looked through some of the issues but not all of them.

**Comment by rozniak** [comment]
Yeah the Start menu needs a fair bit of finishing up... :stuck_out_tongue: 

First points about those programs - at the moment the MFU list is just the first 6 items from listing all desktop entries. It does need a bit more tuning to hide non-programs like the screensavers you're seeing there.

The file responsible for what shows up is here: https://github.com/rozniak/xfce-winxp-tc/blob/master/shell/taskband/res/all.menu

If you have a look in your `/usr/share/applications`, you'll see all the `.desktop` files for things installed on your system - that's what this menu file queries for to find programs. If you open the `.desktop` file for *Binary Horizon* in a text editor, have a look at what `Categories` it has. It might be a screensaver or something that needs adding to `all.menu` under `<Exclude>` . Hopefully that makes some sense, let me know if it doesn't, I can try to explain it better.

----

>The start menu can't seem to find the icon for Seamonkey, a browser I am using for my default email client (though the browser is based on Firefox ~56 the last update it received was on September of this year, so I don't think it's ancient and what's causing the missing icon).

Not entriely sure off the top of my head... it doesn't really do anything fancy to source the icon. I'll have to try and reproduce it my end to see why it can't find it.

> The taskbar doesn't adjust when changing the resolution, it remains where it was before. Killing and rerunning wintc-taskband fixes it.

This is #220 - I don't think it would take too much effort to sort, though need to consider docking the taskband on other edges of the screen first I think - and the usual X11/Wayland fun this involves.

> Changing between themes can cause the All Program menu(s) to keep jittering up and down (and/or add arrows for scrolling despite the menus having enough space). It usually takes just opening and reopening the menu to fix it, but it can be more stubborn depending on what theme was changed and what theme was applied (it doesn't fix in one go when changing from Windows XP Blue to Professional). Killing and rerunning wintc-taskband fixes it.
>
>The start menu's usually offset when opening it for the first time after changing the theme (and it fixes itself when you close it) but that's really minor and you probably knew that already.

This has annoyed me for a long time... it's not a super simple fix because whilst I can explicitly destroy the submenus and stuff on the Start menu to prevent the jittery behaviour, it's a general GTK issue. It affects all popup menus that have already spawned and have had their size changed. Short of forking GTK3 itself I'm not sure there's a clean fix for it, unless I kept identical menu sizes between themes, which would probably not look good.

Start menu itself is a window which I may be able to resolve... atm it's kind of crude the way it pops up and moves over. Kind of a holdover from when it was an XFCE panel plugin :dizzy_face: I can probably fiddle around with `realize` and `map-event` signals to get it to work better when changing themes. :thinking: 

> When I launch XFCE desktop settings (from the menu I get by right clicking on the desktop) with the Windows XP Cursor themes from this TC, it briefly shows the Adwaita black cursor (which I think is the XFCE default) with it's loading animation. However, if I load a page in a browser for example, I do see the normal Windows XP cursor loading animation. The Chicago95 cursors don't have the Adwaita cursor pop in when I open the desktop settings. Again, something really minor but I noticed it.

It's weird, I'm sure I have mapped every cursor name under X11 that I can find... I'll have a poke around in case there's some magical name that sorts it.

**Comment by rozniak** [comment]
I have just found a solution for the cursor problem, will push in a mo.

**Comment by someguy16** [comment]
I went to /usr/share/applications, managed to find the .desktop file for Abstractile. Turns out it was part of xscreensaver, Binary Horizon and Binary Ring seem to have been screensavers as well.

Now there's another program with no icon, Bookmark Editor version 22.12.3 from KDE. That's probably a version from apt (I am using Debian 12 by the way), as I didn't install it on purpose (so it likely got bundled while I was installing something else).

Here's a link to the .deb file for the Seamonkey version I'm using: https://sourceforge.net/projects/ubuntuzilla/files/mozilla/apt/pool/main/s/seamonkey-mozilla-build/seamonkey-mozilla-build_2.53.19-0ubuntu1_amd64.deb/download

Hope these programs help you in debugging the icon problem.

Note: When I first applied the plymouth boot screen, it didn't work, but I got it to work eventually. I think I made it work by running update-grub? Too bad my computer boots up so fast that it only does the fade-in (there's probably a way to prolong the boot screen).

**Comment by someguy16** [comment]
Oh, and the Silver theme makes the text of the icons on my desktop black (I am not using wintc-desktop, though you probably know that because it has no icons implemented yet, right?).

**Comment by rozniak** [comment]
@someguy16 Just fixed the icon issue with SeaMonkey in https://github.com/rozniak/xfce-winxp-tc/commit/3c6ae07ae74c5c1a8db182fe26e535b18c54f9d0

**Comment by someguy16** [comment]
Awesome, thanks! But another question, I have one installed program which is in the "Other" category in the XFCE Applications Menu (& Whisker Menu), but that category's not there at all in the All Programs menu (and the program isn't in any of the other categories either).

I also have some programs that are listed in both "Games" and "System" in XFCE (...I'm not sure why), but these programs don't appear under "System" in the All Programs menu (but do appear in "Games").

**Comment by rozniak** [comment]
What program is it, and are you able to have a look at its desktop entry as before to find out what categories it lives in?

**Comment by someguy16** [comment]
Oh yeah, I had a feeling I should've mentioned the programs but then I went to sleep...

The program in the Other category that doesn't show up in All Programs is Cisco Packet Tracer. Looking at it's .desktop file (I'll upload all of them) it doesn't have _any_ category whatsoever. You probably won't need to download it but if you do try, beware that their website sucks ass (which is the only official place to get the .deb file from as far as I know, but there's probably some software website that provides it).

The games that show up in both Games and System in _XFCE_ but show up only in Games in All Programs are desmume and mednaffe (both installed from apt, apt-get install desmume mednaffe). Their categories are GTK, Game, and Emulator (desmume additionally has GNOME, mednaffe doesn't have GNOME), maybe XFCE groups them in System because they're emulators? But I'm not sure why they aren't under System here...

.desktop files:
Cisco Packet Tracer:
[cisco-pt821.txt](https://github.com/user-attachments/files/18280864/cisco-pt821.txt)

Desmume (it has two variants of the program):
[desmume.txt](https://github.com/user-attachments/files/18280874/desmume.txt)
[desmume-glade.txt](https://github.com/user-attachments/files/18280875/desmume-glade.txt)

Mednaffe: 
[mednaffe.txt](https://github.com/user-attachments/files/18280884/mednaffe.txt)

**Comment by rozniak** [comment]
Man it's been ages since I last used Packet Tracer, doesn't surprise me that they didn't bother to fill in the categories. I'll have a fiddle around with it. :zany_face: 

> (And also, does taskband "build" the menus itself? I am only asking because I am curious about how that works.)

The Start menu uses `libgarcon` which is an XFCE library for handling XDG menu files. So it 'roughly' works the same way as the XFCE Application Menu, only it uses `.menu` files written by me to try and group things a little closer to how Windows sort of would (with *Accessories* or *System*).

It's kind of a pain honestly, in general it's a stopgap solution until I write a proper Start menu that behaves more closely to Windows XP instead of just using the XDG menu stuff.

The files responsible for defining the layouts are:
`/shell/taskband/all.menu` - For the MFU list (top 6 items) (crap)
`/shell/taskband/applications.menu` - For the *All Programs* popup menu

**Comment by rozniak** [comment]
There should now be an *Other* category with Packet Tracer in it as of https://github.com/rozniak/xfce-winxp-tc/commit/1cc1d284f257cdf7bd48e968a75c5fcd5baff052

**Comment by someguy16** [comment]
Nice!

**Comment by someguy16** [comment]
Nice! What about the Silver theme issue? (No rush on fixing, but a reminder just in case I buried it too far)

**Comment by rozniak** [comment]
I will have to look at that when I'm back at my machine again, currently out at the moment. :stuck_out_tongue:

**Comment by someguy16** [comment]
Okay uh, I hope I'm not being annoying (unless I had already become annoying a while ago):

![Screenshot_2025-01-03_19-15-57](https://github.com/user-attachments/assets/f95d0ba4-e557-4482-9dc1-bd71287f9b6d)
![Screenshot_2025-01-03_19-16-26](https://github.com/user-attachments/assets/0a480067-65b6-4084-8b8e-ca3d761b2833)
![Screenshot_2025-01-03_19-16-44](https://github.com/user-attachments/assets/9742ab43-186c-49e7-86a7-0188be628359)

This happens. Open too many windows and they intrude into the system tray a bit. This is only a problem with the Whistler (Professional) and Windows Classic themes. This doesn't happen on the Luna or Silver theme.

Happy new year by the way.

**Comment by rozniak** [comment]
Yeah I have noticed that before but never got around to making an issue for it. I wasn't entirely sure whether it happens after switching themes or whether it is actually just a size allocation issue... could do with reliable repro steps. :thinking:

**Comment by someguy16** [comment]
To make sure this wasn't due to changing the theme, I applied the classic theme, killed taskband, reran it, opened a dozen windows and they still protrude from the system tray (I also logged out and logged back in, still does that).

I forgot to mention though, either the most right window can be seen through the system tray (but icons and borders render over it), or it can't be seen (but it's still clear that the borders extend behind the system tray). This is a bit inconsistent? I'm not sure how to make one "happen"

![image](https://github.com/user-attachments/assets/61ab068a-92d5-4be3-a8af-225bbc296fe3)
![image](https://github.com/user-attachments/assets/9c5f0a68-9a60-46c7-a08a-c351a78a535a)

**Comment by rozniak** [comment]
I just had a thought I might know what it is, potentially my size allocate implementation is broken because it doesn't account for margins between buttons - Classic and Professional have margins, the Luna themes don't.

At the moment I'm working on Explorer a bit for #413 but will try to look afterwards.

**Comment by someguy16** [comment]
I changed the line 4390 in the the gtk.css file in the folder gtk-3.0 of the Silver Theme folder, and that seems to have fixed the black icon labels issue? ...It's probably not as simple as that, but I don't know... everything seems to look fine.

Original line: @define-color theme_selected_fg_color #000000;
Changed line: @define-color theme_selected_fg_color #FFFFFF;
(yeah i know i just changed the color from black to white)

Here's a before and after:
![image](https://github.com/user-attachments/assets/c05a2d59-850c-44cb-8bec-d5851a879702)
![image](https://github.com/user-attachments/assets/05a975c1-6917-4e6c-b282-917338bbf91d)

**Comment by rozniak** [comment]
Hmmm... it's kind of stupid that it's using the selected colour scheme when the items are not selected. Perhaps it is worth checking the upstream xfdesktop source for this... :thinking:

**Comment by rozniak** [comment]
Continue this specific xfdesktop issue in #444
